### PR TITLE
Added IDImpl struct

### PR DESCRIPTION
**Pull request by moltzaum**
## What does this PR (Pull Request) do?

IDImpl is a struct that implements GetKeys and SetKeys for the common case where the only key is a single numerical ID. The idea is to make a single, common implementation that can be used across all endpoints that can use it. As a bonus these endpoints don't need to make a GetKeys or SetKeys function, nor do they have to provide the json and db tags and we know the behavior we want is implemented.

The downside is that current swagger comments may be broken (although we weren't exactly using it anyway).

This is mostly a test to get feedback. I've had an idea like this for awhile, but didn't implement it because I had doubts. I think it is relatively useful, although it is debatable whether or not this works well as a layer of abstraction since "every abstraction layer has a cost". I was considering other ways to do the same thing in a better way. Also, some might perfer to keep the tc structs 'clean'.

For now, I've only made the CDN struct use the IDImpl. IDImpl only works for Nullable structs (the structs that we define endpoints on).

## Which Traffic Control components are affected by this PR?

The main affected component is Traffic Ops, but since I've edited a TC struct, it could affect any component that uses the structs. go is smart enough so that `StructType.EmbeddedType.Field` can be called as `StructType.Field` as long as that field name doesn't exist higher up in the embedded struct hierarchy, so moving the field to an embedded type shouldn't affect much.

## What is the best way to verify this PR?

Tests already exist for CDNs and should pass.

## The following criteria are ALL met by this PR

- [x] Tests already exist for affected pieces of code
- [ ] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3681/
Test FAILed.

**Comment by moltzaum** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3682/
Test FAILed.

File: lib/go-tc/cdns.go [hunk 1]
```diff
@@ -83,7 +85,7 @@ type CDNNullable struct {
 	// ID of the CDN
 	//
 	// required: true
-	ID *int `json:"id" db:"id"`
+	impl.ID
```
Comment (ocket8888) [inline lib/go-tc/cdns.go]: `lib/go-tc/cdns.go:88:2: error: reference to undefined identifier 'impl.ID'`
I think either this should be `impl.ImplID` or you need to rename the struct to `ID`. Either will work, but I'm partial to the latter, myself; `impl.ImplID` feels redundant.

File: traffic_ops/traffic_ops_golang/cdn/cdns.go [hunk 2]
```diff
@@ -58,14 +58,6 @@ func (cdn TOCDN) GetKeyFieldsInfo() []api.KeyFieldInfo {
...
 
-//Implementation of the Identifier, Validator interface functions
-func (cdn TOCDN) GetKeys() (map[string]interface{}, bool) {
-	if cdn.ID == nil {
-		return map[string]interface{}{"id": 0}, false
-	}
-	return map[string]interface{}{"id": *cdn.ID}, true
-}
-
 func (cdn TOCDN) GetAuditName() string {
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/cdn/cdns.go]: This `func` accesses the `cdn.ID` field, which no longer has the right type, causing compilation issues.

**Comment by moltzaum** [comment]
Closing, because I don't think this will get merged and it was experimental anyway.

**Pull request closed without merging**